frame/collective/src/lib.rs

```diff
@@ -704,10 +727,132 @@ decl_module! {
+		///
+		/// Since the collective may want to be able to represent a different ratio than exists,
+		/// we allow the re-dispatch to specify the target ratio, as long as it is less restrictive
+		/// than the provided origin.
+		///
+		/// Can only be called by the collective origin.
+		#[weight = {
+			T::WeightInfo::dispatch_as_quantity_account(*length_bound)
+			.saturating_add(proposal.get_dispatch_info().weight)
+		}]
+		fn dispatch_as_quantity_account(origin,
+			proposal: Box<T::Proposal>,
+			#[compact] length_bound: u32,
+		) -> DispatchResult {
+			let maybe_raw_origin: Result<RawOrigin<T::AccountId, I>, _> = <T as Config<I>>::Origin::from(origin).into();
+
+			let raw_origin = match maybe_raw_origin {
+				Ok(origin) => origin,
+				Err(_) => return DispatchError::BadOrigin.into(),
+			};
+
+			let proposal_len = proposal.using_encoded(|x| x.len());
+			ensure!(proposal_len <= length_bound as usize, Error::<T, I>::WrongProposalLength);
+
+			let account = Self::origin_to_quantity_account(raw_origin)?;
+			let proposal_hash = T::Hashing::hash_of(&proposal);
+			let signed_origin = frame_system::RawOrigin::Signed(account.clone());
+			let result = proposal.dispatch(signed_origin.into());
+			Self::deposit_event(
+				RawEvent::AccountExecuted(account, proposal_hash, result.map(|_| ()).map_err(|e| e.error))
+			);
+			Ok(())
+		}
 	}
 }
 
 impl<T: Config<I>, I: Instance> Module<T, I> {
+	/// A simple helper function to generate a collective ratio account from the collective origin.
+	///
+	/// A ratio account is an account which represents a ratio of n / d users where n is the number
+	/// of approving members, and d is the total number of members. n and d can be any fraction which
+	/// is less or equally restrictive than the origin.
+	pub fn origin_to_ratio_account(
+		origin: RawOrigin<T::AccountId, I>,
+		target_n: u32,
+		target_d: u32,
+	) -> Result<T::AccountId, DispatchError> {
+		ensure!(target_n <= target_d, Error::<T, I>::NotEnoughMembers);
+		match origin {
+			RawOrigin::Members(n, d) => {
+				// n is the number of members who voted aye. d is the total number of members.
+				// Should never be able to make an origin which represents more members than exists.
+				ensure!(target_d <= d, Error::<T, I>::NotEnoughMembers);
+				// The target ratio should be less than or equal to the ratio provided by the origin.
+				// For example 6/13 cannot create a 1/2 account, but 7/13 can.
+				// To check (target_n / target_d) <= (n / d) we check (target_n * d) <= (target_d * n)
+				let target_n_d = target_n.checked_mul(d).ok_or(Error::<T, I>::Overflow)?;
+				let target_d_n = target_d.checked_mul(n).ok_or(Error::<T, I>::Overflow)?;
+				ensure!(target_n_d <= target_d_n, Error::<T, I>::NotEnoughMembers);
+				// Create the account.
+				Ok(T::ModuleId::get().into_sub_account((target_n, target_d)))
+			},
+			RawOrigin::Member(who) => {
+				Ok(T::ModuleId::get().into_sub_account(who))
+			},
```

Comment:
I was expecting this to also be an error. What is the rationale?

Reply:
Yeah probably should be an error. I was just able to handle it so we do.